appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/CommitDelayingResponse.java

```diff
@@ -77,9 +77,6 @@
    */
   public CommitDelayingResponse(HttpServletResponse response) throws IOException {
     super(response);
-    // Set the buffer size explicitly, as this ensure both that jetty's aggregate
-    // buffer is the correct size and that the commit threshold is that size.
-    response.setBufferSize(CommitDelayingOutputStream.MAX_RESPONSE_SIZE_BYTES);
```

Comment:
It may be better to replace this line with:
```
    response.setBufferSize(response.getBufferSize());
```
as this will reset the the commit threshhold (by default 1/4 of the buffer size) to be the buffer size.    By default if the buffer size is 4K, then any individual write larger  than 1K will be handled with a direct write rather than aggregation into the buffer.

Note also that the default buffer size is 32KB, so it may be worth a little memory to increase this to say 128KB to ensure that more resources are committed after generation.     The  prior 32MB size was obviously way too large.

Note that the output stream wrapper is not really doing anything and could pretty much be removed.   It currently will cause a hard write failure at 32MB, but if the response is already committed and being sent over a streaming transport, then there is little reason to have the hard fail.

Alternately, we could re-implement the original GAE implementation that had growing buffers that could grow the 32MB before failure.